src/SFA.DAS.Employer.PR.Web/Controllers/SelectLegalEntityController.cs

```diff
@@ -84,4 +87,15 @@ private void SetSessionModel(SelectLegalEntitiesViewModel currentModel)
 
         return model;
     }
+
+    private async Task SetSessionForSelectionMade(string employerAccountId, long legalEntityId, CancellationToken cancellationToken)
+    {
+        var response = await _outerApiClient.GetAccountLegalEntities(employerAccountId, cancellationToken);
+
+        var numberOfLegalEntities = response.AccountLegalEntities.Count;
+        foreach (var legalEntity in response.AccountLegalEntities.Where(l => l.Id == legalEntityId))
+        {
+            SetSessionModel(legalEntity.Id, legalEntity.Name, numberOfLegalEntities);
```

Comment:
Why are we setting session model for each legal entity?

Reply:
it's only one, but its not clear or readable (and not clean code) so changing this to be on the first match.